mods/cnc/maps/cnc64gdi01/cnc64gdi01.lua

```diff
@@ -0,0 +1,205 @@
+
+AutocreateSquads = { auto1, auto2, auto3, auto4, auto5, auto6, auto7, auto8 }
+
+AttackTriggers = { AttackTrigger1, AttackTrigger2, AttackTrigger3, AttackTrigger4 }
+
+harvester = { "harv" }
+
+SamSites = { SAM01, SAM02 }
+Obelisk = { Obelisk01 }
+BiotechCenter = { Biolab }
+
+WorldLoaded = function()
+	player = Player.GetPlayer("GDI")
+	enemy = Player.GetPlayer("Nod")
+
+	Trigger.OnObjectiveAdded(player, function(p, id)
+		Media.DisplayMessage(p.GetObjectiveDescription(id), "New " .. string.lower(p.GetObjectiveType(id)) .. " objective")
+	end)
+	Trigger.OnObjectiveCompleted(player, function(p, id)
+		Media.DisplayMessage(p.GetObjectiveDescription(id), "Objective completed")
+	end)
+	Trigger.OnObjectiveFailed(player, function(p, id)
+		Media.DisplayMessage(p.GetObjectiveDescription(id), "Objective failed")
+	end)
+
+	Trigger.OnPlayerWon(player, function()
+		Media.PlaySpeechNotification(player, "Win")
+	end)
+
+	Trigger.OnPlayerLost(player, function()
+		Media.PlaySpeechNotification(player, "Lose")
+	end)
+
+	destroySAMsCenterObjective = player.AddPrimaryObjective("Destroy the SAM sites protecting the Obelisk.")
+	destroyObeliskObjective = player.AddPrimaryObjective("Destroy the Obelisk.")
+	destroyBiotechCenterObjective = player.AddPrimaryObjective("Destroy the biotech facility.")
+
+	Trigger.OnAllKilled(SamSites, function()
+		AirSupport = Actor.Create("airstrike.proxy", true, { Owner = player })
+		AirSupportEnabled = true
+		player.MarkCompletedObjective(destroySAMsCenterObjective)
+	end)
+
+	Trigger.OnDamaged(Obelisk01, function()
+		Trigger.AfterDelay(DateTime.Seconds(1), Obelisk01.Kill)
+	end)
+
+	Trigger.OnAllKilled(Obelisk, function()
+		player.MarkCompletedObjective(destroyObeliskObjective)
+		Trigger.AfterDelay(DateTime.Seconds(5), function() Reinforce(MCVReinforcements) end)
+		ObeliskFlare.Destroy()
+		if AirSupportEnabled then AirSupport.Destroy() end
+	end)
+
+	Trigger.OnAllKilled(BiotechCenter, function()
+		player.MarkCompletedObjective(destroyBiotechCenterObjective)
+	end)
+	
+	Trigger.OnCapture(Biolab, function()
+		Biolab.Kill()
+	end)
+
+	Trigger.OnDamaged(Biolab, HuntTriggerFunction)
+	
+	AIRepairBuildings(enemy)
+	AIRebuildHarvesters(enemy)
+
+	Utils.Do(AttackTriggers, function(a)
+		Trigger.OnKilledOrCaptured(a, function()
+			NodVehicleProduction(Utils.Random(AutocreateSquads))
+		end)
+	end)
+
+	Trigger.AfterDelay(DateTime.Seconds(150), AutoCreateTeam)
+
+	NodInfantryProduction()
+
+	Camera.Position = UnitsRally.CenterPosition
+	ObeliskFlare = Actor.Create('flare', true, { Owner = player, Location = Flare.Location })
+	Reinforce(CommandoReinforcements)
+end
+
+Tick = function()
+	if DateTime.GameTime > DateTime.Seconds(5) and player.HasNoRequiredUnits() then
+		player.MarkFailedObjective(destroyBiotechCenterObjective)		
+	end
+end
+
+Reinforce = function(units)
+	Media.PlaySpeechNotification(player, "Reinforce")
+	ReinforceWithLandingCraft(units, lstStart.Location, lstEnd.Location, UnitsRally.Location)
+end
```

Comment:
Actually, we need to do it manually in GDI01 and 02 because `ReinforceWithTransport` does not play well with the open-topped landing crafts.  Fixing that generally would make the function unnecessarily complicated compared with faking it in lua in the missions that need it.